nums/core/array/utils.py

```diff
@@ -26,6 +26,37 @@
 # pylint: disable = no-member, trailing-whitespace
 
 
+def find_output_blocks(X_blocks, total_elements):
```

Comment:
Can we rename this to `find_diag_output_blocks`?

Reply:
Yes done.